pkg/app/pipectl/cmd/application/sync.go

```diff
@@ -0,0 +1,114 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package application
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pipe-cd/pipe/pkg/app/pipectl/client"
+	"github.com/pipe-cd/pipe/pkg/cli"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+type sync struct {
+	root *command
+
+	appID         string
+	status        []string
+	checkInterval time.Duration
+	timeout       time.Duration
+}
+
+func newSyncCommand(root *command) *cobra.Command {
+	c := &sync{
+		root:          root,
+		checkInterval: 15 * time.Second,
+		timeout:       5 * time.Minute,
+	}
+	cmd := &cobra.Command{
+		Use:   "sync",
+		Short: "Sync an application.",
+		RunE:  cli.WithContext(c.run),
+	}
+
+	cmd.Flags().StringVar(&c.appID, "app-id", c.appID, "The application ID.")
+	cmd.Flags().StringSliceVar(&c.status, "wait-status", c.status, fmt.Sprintf("The list of waiting statuses. Empty means returning immediately after triggered. (%s)", strings.Join(availableStatuses(), "|")))
+	cmd.Flags().DurationVar(&c.checkInterval, "check-interval", c.checkInterval, "The interval of checking the requested command.")
+	cmd.Flags().DurationVar(&c.timeout, "timeout", c.timeout, "Maximum execution time.")
+
+	cmd.MarkFlagRequired("app-id")
+
+	return cmd
+}
+
+func (c *sync) run(ctx context.Context, t cli.Telemetry) error {
+	statuses, err := makeStatuses(c.status)
+	if err != nil {
+		return fmt.Errorf("invalid deployment status: %w", err)
+	}
+
+	cli, err := c.root.clientOptions.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize client: %w", err)
+	}
+	defer cli.Close()
+
+	deploymentID, err := client.SyncApplication(ctx, cli, c.appID, c.checkInterval, c.timeout, t.Logger)
+	if err != nil {
+		return err
+	}
+
+	t.Logger.Info(fmt.Sprintf("Successfully triggered deployment %s", deploymentID))
+	if len(statuses) == 0 {
+		return nil
+	}
+
+	t.Logger.Info("Waiting until the deployment reaches one of the specified statuses")
+
+	return client.WaitDeploymentStatuses(
+		ctx,
+		cli,
+		deploymentID,
+		statuses,
+		c.checkInterval,
+		c.timeout,
+		t.Logger,
+	)
+}
+
+func makeStatuses(statuses []string) ([]model.DeploymentStatus, error) {
+	out := make([]model.DeploymentStatus, 0, len(statuses))
+	for _, s := range statuses {
+		status, ok := model.DeploymentStatus_value["DEPLOYMENT_"+s]
+		if !ok {
+			return nil, fmt.Errorf("bad status %s", s)
+		}
+		out = append(out, model.DeploymentStatus(status))
+	}
+	return out, nil
+}
+
+func availableStatuses() []string {
+	out := make([]string, 0, len(model.DeploymentStatus_value))
+	for s := range model.DeploymentStatus_value {
+		out = append(out, strings.TrimPrefix(s, "DEPLOYMENT_"))
+	}
+	return out
+}
```

Comment:
nit: move this part to `model/deployment.go` for using across cli's commands 👀

Reply:
I still think we should keep it inside this `pipectl` scope.
(The model package contains the code that can be used by any packages in the PipeCD project, while this function (or status naming) is just for this case.)

Reply:
What there are duplicated functions is a bit strange, but this is too tiny and domain-specific to make it a common public function. I agree to keep those as is 👍

Reply:
I understand your thought and don't have a strong opinion on this, but to me, making a string by stitching enum values is good enough to be stuck with the enum itself. Otherwise, we should find another place to place this part of code since it will be shared across any newly added commands, if I guess right 😅

Reply:
let's move this part of code to a suitable location latter if necessary 😅

Reply:
IMO, these helpers will only be used in the child commands of `application` and `deployment`, so I guess no more copy for any newly added commands.
(Putting them in a new package inside this `pipectl` is not good too because of naming anti-patterns such as `util` or `helper`.)

I agree with you that we should keep using the enum string for CLI too. By that, we can avoid those duplications, avoid the confusion about the status name. The consistency is important.

So let me make another PR to fix this thing.

Reply:
> So let me make another PR to fix this thing.

thanks 🙆